### DISCUSSION TITLE
About freeplane on mobile

### DISCUSSION THREAD

**Discussion by bepolymathe** [freeplane/freeplane#480] [General]
Hi all, 

Out of curiosity, do you have a proper solution for reading maps from the phone?
😃 
I don't need this feature often but now that I'm trying to manage my tasks in freeplane again it could be useful...

**Comment by ldmpub** [comment]
Hi,

This was one of many topics discussed in our last community meeting ([link to the notes](https://github.com/freeplane/freeplane/discussions/339#discussioncomment-2842778))

If your device runs Android the best solution yet is to use DroidPlane but if you found other / better solution please share it ...

**Reply by ldmpub** [reply]
Extract of the meeting minutes on this particular topic:
![image](https://user-images.githubusercontent.com/1452146/176958489-144610ba-cc55-4865-ba94-2587c5b43184.png)

**Reply by bepolymathe** [reply]
Thanks for your feedback. I'm glad to hear that there are plans to do so. I'm still amazed that the best solution is the old Droidplane. 
In any case, I understand that it is a big challenge in terms of development for the community. 
Thanks for your feedback.

**Comment by 1-DimensionalBox** [comment]
Indeed, having the Freeplane community develop a mobile phone application is currently not possible, unfortunately.

For what it's worth; I am currently looking into exporting to and importing information from the free and open source [Orgzly](http://www.orgzly.com) app. I will let you know if it works out!

**Reply by ldmpub** [reply]
Thanks for the info and good work in this direction...

**Reply by kha84** [reply]
Hello there. Any updates on that?

**Reply by ThAugustinStatsLMU** [reply]
Dear all, 
Has someone had some experience in using Freeplane under the rollApp environment on an Android mobile phone? Freeplane is listed as a compatible program on the rollApp webpage. I was able to view some maps using the Android version of Chrome, but not able at all to edit the maps properly. 
Many thanks and best wishes,
Thomas

**Comment by tino1003870** [comment]
Develoling my ideas and plans in "HalnaMind" and "HalnaOutliner" on Android, I can export and import FreePlane. 
I never worked with attributes in this project stage and I never tried to sync somehow. But it is imagible to make it work using some Nextcloud option.
BR, Tino

**Reply by ldmpub** [reply]
Hi, thanks to point us to Halnamind Android app 👍.

Unfortunatly, after a Freeplane map import, nodes with icons are displayed as `(Untitled)` 🙃.

May I suggest to write a documentation page based on our tests to find alternatives to "no Freeplane Android app" ?

**Comment by ldmpub** [comment]
I have discovered an other (unfortunately, discontinued) Android app to view and edit freeplane mind maps: **Z Mind**, see https://zmindapp.com

- Pro: Can display Freeplane maps as a **collapse / expand mind map layout**

- Cons: **Word search in nodes is not available** (it has been added recently to DroidPlane 👍)

**Reply by euu2021** [reply]
Interesting. Unfortunately, it has a lot of bugs and it doesn't work for me.

I'm now more interested in editing FP maps in Android. As an alternative, I'm currently using the Dynalist app, and I find the UI/UX ideal for the phone small touch screen (it displays as an outliner). And it has good exporting options that can be used to export branches to FP. I wish someday we have an app like that to make simple edits in FP maps 😏

**Comment by smellymoo** [comment]
I was looking into options for making freeplane work on android. If freeplane is C/C++, which I would hope so, then QT has really developed since I last checked, and is now really the only option for true cross-platform C/C++ applications. Here I am hoping freeplane isn't (terrible) Java.

Any news on android development. Still no plan?

**Reply by HaaiHenkie** [reply]
Freeplane is Java based. I got it running on a Android tablet by installing Termux and the experimental Termux X server. It needs some more explaining, so if people are interested I could write down the steps in a separate post. I won't work on a phone since it will be to small to read. I think the main problem is that Freeplane is not designed for such a small screen.

Years ago there was an app that was compatible with Freemind and was surprisingly good in editing mind maps on a phone. I forgot its name. It could also do Freeplane but you would lose the Freeplane specific functionality of your map. Unfortunately Mindmanager took over that app and after a while it just disappeared.

**Reply by dpolivaev** [reply]
As a mind map editor for smartphones I personally use https://simplemind.eu/

**Reply by HaaiHenkie** [reply]
The app I was talking about was Thinking Space. I found an old page with some screenshots: https://download.cnet.com/thinking-space/3000-2124_4-75325500.html

It was acquired by Mindjet in 2011 and released as Mindjet for Android. The quality/functionality was not as good as Thinking Space. Nowadays it is called MindManager Go and you can only read files created with the MindManager desktop application.

How to effectively kill a nice app...

**Comment by ldmpub** [comment]
@dpolivaev thank you for the tip 👌.

Could you please give us little more details:
* **Which version** of https://simplemind.eu/ do you use? (the lite / free version or the paid one?)
* Have you experienced a **personnal method to  work on Freeplane mindmaps in SimpleMind**? _For instance using mindmaps export / import and taking care not to use particular node formats or feature in each mindmaps editor?_

**Reply by dpolivaev** [reply]
I purchased pro version for Android about 10 years ago

**Reply by lumasci** [reply]
There seem to be a viewer version of simpleMind on Mac, iPad, iPhone, Windows but android
https://simplemind.eu/download/simplemind-viewer/

**Reply by ldmpub** [reply]
Still searching a free (or not so expensive) solution to "display" Freeplane Mind maps on my Android phone ...

DroidPlane has currently a bug preventing to see the first node ... at each level 🙁!

**Comment by smellymoo** [comment]
@HaaiHenkie oh, if it's Java, then porting it to android is actually a lot easier. If the big issue is the text of the mindmap won't be readable, then pinchzoom would fix that. As the developer of https://drawsum.com, I can easily help with that small part of the port. The other part would be adapting the UI, which is relatively easy with Android Studio.

Seems like it's probably time port to android, most people use their phone 90% and their laptops 10%, so it would be a great popularity boost to freeplane to be usable on both.

@dpolivaev thanks, I'll try it out. Trying to find an open source free option, looks difficult.

**Reply by nnako** [reply]
Hi @smellymoo , very interesting project. But `drawsum` is no open source, eh? I saw a project with the same name on GoogleCode, but there was no source.

**Reply by marquantl** [reply]
It would be great @smellymoo. Then would be presented as an image? or with interactive links?
[Sync-in](https://sync-in.com/) might be a good syncing choice.

**Comment by kha84** [comment]
I was also very keen to see some kind of application tandem for Linux + Android I can use. After months of experimenting with different software I ended up using Obsidian for both desktop and mobile plus Syncthing for syncing. Yes, it's not FOSS (it's just free, but not open source) but  it has a canvas feature, that can be used for mindmap purposes. 
Sorry for breaking-in. I know it's not a proper solution, but probably someone who is desperate can use this as workaround.

**Reply by marquantl** [reply]
@kha84 you would have freeplane as a plugin?

**Reply by kha84** [reply]
To be honest, I'm probably not the target audience already, provided I have found all-in-one solution in the form of Obsidian that works on all platforms I need (Linux, Android, Windows, iOS). 

BUT, in case if and when things go south with Obsidian and I have to migrate from it elsewhere, having a Freeplane as a plugin to a software like [Logseq](https://logseq.com/) / [Zettlr](https://www.zettlr.com/) / [Joplin](https://github.com/laurent22/joplin) / [marktext](https://github.com/marktext/marktext) / [Folio](https://github.com/toolstack/Folio) / [Paper](https://gitlab.com/posidon_software/paper) / [zim](https://www.zim-wiki.org/) / [Notable](https://github.com/notable/notable) / [Anytype](https://anytype.io/) / [Notesnook](https://notesnook.com/) / [StandardNotes](https://github.com/standardnotes/app) / [Apostrophe](https://apps.gnome.org/Apostrophe/) / [Klevernotes](https://apps.kde.org/klevernotes/) / [IOTAS](https://apps.gnome.org/en/Iotas/) / [qownnotes](https://github.com/pbek/QOwnNotes) or alike will be considered as a big benefit of  that software, as most of them are  tend to miss the mind map features.
